mmengine/hooks/early_stopping_hook.py

```diff
@@ -0,0 +1,108 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import warnings
+from typing import List, Optional, Union
+
+from mmengine.registry import HOOKS
+from .hook import Hook
+
+DATA_BATCH = Optional[Union[dict, tuple, list]]
+
+
+@HOOKS.register_module()
+class EarlyStoppingHook(Hook):
+    """Early stop the training when the metric reached a plateau.
+
+    Args:
+        metric (str): The metric key to decide early stopping.
```

Comment:
Referring to https://keras.io/api/callbacks/early_stopping/, maybe `monitor` will be a more proper parameter name? Maybe someday MMEngine will support calculate the validation loss in the train loop :rofl: